java/src/main/native/CMakeLists.txt

```diff
@@ -13,7 +13,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #=============================================================================
-cmake_minimum_required(VERSION 3.12 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
```

Comment:
I think the normal cudf contributing doc says 3.12+, do we need 3.14 here?

Reply:
The `FetchContent` module was added in CMake 3.14.

Reply:
If this requires CMake 3.14 then the conda environment yml files need to be updated accordingly.  They currently spec >= 3.12.  Otherwise we risk having a non-CI build (e.g.: from developers) fail during the JNI build step.

Alternatively the explicit fetching of spdlog could be toggled by a cmake flag (or even better, auto-detected via the environment if possible by looking for `CONDA_PREFIX`?).  Then cmake 3.12 is sufficient when using conda and spdlog won't be wastefully fetched in that environment.

Reply:
Thinking about this more, I'm wondering if this is the appropriate place to fix it.  IIUC this is an issue with a CI environment that is building the JNI bindings without conda.  Therefore it will need to fetch RMM and install it separately because conda will not be used to fetch it.  spdlog is only needed because RMM headers need it.  Therefore it seems to me a better fix would be to update the CI scripts/makefiles to install the spdlog headers when it installs RMM in preparation for the cudf build.

That localizes the change to the place that needs it, and as a bonus we're not fetching spdlog in cases where it doesn't make sense (e.g.: a dev build that is using the conda env as expected).

Reply:
At least for RMM, the desire is to not use conda for c++ dependency management. For PTDS to work, I need to pull in thrust/cub through FetchContent as well.

Reply:
If that's the way we want to go then this change should not be Java-specific.  Clearly C++ dependency management is relying on conda for a lot of stuff today (like RMM headers).  If we want to change that then that's way beyond the scope of this PR and needs to involve a lot more discussion from project stakeholders.

IMO in the short term the more straightforward, cleaner fix is to have the CI environment that is installing RMM outside of conda to also install spdlog.  Having it (redundantly) install it here is not clean.

Reply:
Actually I've recently added spdlog as a dependency for the librmm conda package. So if the CI installs the nightly rmm conda package, it should have gotten spdlog. I'm not sure why the nightly build is failing. I was told it's not using conda.